internal/server/server.go

```diff
@@ -420,37 +429,51 @@ func (s *Service) Run() error {
 func (s *Service) registerAzureComponents() {
 	logger.Infof("Registering Azure Components...")
 
-	// Conditional registration based on UseLegacyTools flag
-	if s.cfg.UseLegacyTools {
-		logger.Infof("Using legacy Azure CLI tools (az_aks_operations, az_compute_operations)")
-		// AKS Operations Component (legacy)
-		s.registerAksOpsComponent()
-	} else {
-		logger.Infof("Using azure-api-mcp tool (call_az)")
-		// Azure API MCP Component (new)
-		s.registerAzureApiComponent()
+	// Azure CLI Component (unified registration)
+	if components.IsComponentEnabled("az_cli", s.cfg.EnabledComponents) {
+		if s.cfg.UseLegacyTools {
+			logger.Infof("Registering Azure CLI Component with legacy tools (az_aks_operations, az_compute_operations)")
+			s.registerAksOpsComponent()
+		} else {
+			logger.Infof("Registering Azure CLI Component with unified tool (call_az)")
+			s.registerAzureApiComponent()
+		}
 	}
 
 	// Monitoring Component
-	s.registerMonitoringComponent()
+	if components.IsComponentEnabled("monitor", s.cfg.EnabledComponents) {
```

Comment:
We are following the earlier code pattern, and we can keep you change in this PR
But I suggest we use a map or something to iterate over the map instead of listing all registrations.

Reply:
EnabledComponents is a list got from user input. It is easy and clear to check it in this way, converting it to a map is okay but I don't see too much benefit. I would like to keep it like this. thaks.